sktime/classification/early_classification/tests/_test_teaser.py

```diff
@@ -66,15 +73,22 @@ def test_teaser_with_different_decision_maker():
     final_decisions = np.zeros(10)
 
     X_test = from_nested_to_3d_numpy(X_test)
-    states = None
+    X_test = X_test[indices, :, :]
+
+    ec_idx = np.arange(0, 10)
+    decisions = np.zeros(len(X_test), dtype=bool)
     for i in teaser.classification_points:
-        X = X_test[indices, :, :i]
-        probas, decisions, states = teaser.predict_proba(X, state_info=states)
+        inv_dec = np.invert(decisions)
+        X_test = X_test[inv_dec, :, :]
+        ec_idx = ec_idx[inv_dec]
 
-        for n in range(10):
-            if decisions[n] and final_decisions[n] == 0:
-                final_probas[n] = probas[n]
-                final_decisions[n] = i
+        X = X_test[:, :, :i]
+        probas, decisions, states = teaser.update_predict_proba(X)
```

Comment:
Does it still need state-info to be returned? I think, decisions and probas are the two essential ones?

Reply:
Ok